Nix build error (missing headers)

### Godot version

4.0.dev(f00803b)

### System information

NIXOS

### Issue description

Header files appear to be missing when building in nixos when using a "flake" (a special type of build file).
Here is the log file:

```
1vs9rp7aqianv6sprz2-godot.drv
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/dsdcfwrj3xc4vc879yvb5yvxcy308g>
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: target=release_debug platform=linuxbsd pulseaudio=tru>
scons: Reading SConscript files ...
Checking for C header file mntent.h... yes
scons: done reading SConscript files.
scons: Building targets ...
^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial b>
^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial b>
In file included from ./drivers/alsa/audio_driver_alsa.h:40,
                 from platform/linuxbsd/os_linuxbsd.h:36,
                 from platform/linuxbsd/godot_linuxbsd.cpp:41:
./drivers/alsa/asound-so_wrap.h:1289:10: fatal error: alsa/asoundl>
 1289 | #include <alsa/asoundlib.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
scons: *** [platform/linuxbsd/godot_linuxbsd.linuxbsd.opt.tools.64>
scons: building terminated because of errors.
[Time elapsed: 00:00:01.426]
...skipping...
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/dsdcfwrj3xc4vc879yvb5yvxcy308g>
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: target=release_debug platform=linuxbsd pulseaudio=tru>
scons: Reading SConscript files ...
Checking for C header file mntent.h... yes
scons: done reading SConscript files.
scons: Building targets ...
^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial b>
^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial b>
In file included from ./drivers/alsa/audio_driver_alsa.h:40,
                 from platform/linuxbsd/os_linuxbsd.h:36,
                 from platform/linuxbsd/godot_linuxbsd.cpp:41:
./drivers/alsa/asound-so_wrap.h:1289:10: fatal error: alsa/asoundl>
 1289 | #include <alsa/asoundlib.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
scons: *** [platform/linuxbsd/godot_linuxbsd.linuxbsd.opt.tools.64>
scons: building terminated because of errors.
[Time elapsed: 00:00:01.426]
```

This appears to be due to the header files not being added to the PATH.
These errors are fixed by the changes to `detect.py` in [this pull request](https://github.com/godotengine/godot/pull/59739) that I created. 
Nixos does not use standard paths like a regular distro, instead it uses a "store" where every package is prefixed by a hash, allowing for multiple instances of the package to exist. 
The `alsa/asoundlib.h` header for example is stored at `/nix/store/4v5n58wzh7qvl9zvszwx442ymla51m3r-alsa-lib-1.2.6.1-dev/include/alsa/asoundlib.h` on my machine.

### Steps to reproduce

This error is easily reproduced on any machine with the nix package manager with flakes enabled. Simply clone my flake repository [here](https://github.com/IllustratedMan-code/Godot-Flake) and run `nix build` inside the directory.

### Minimal reproduction project

_No response_

Looks like Xfixes is still required.
Here is the log
```@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/dsdcfwrj3xc4vc879yvb5yvxcy308g>
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
applying patch /nix/store/5zm9z3bfal5asx9xbc1lw9phjrimnjya-godot-p>
patching file platform/linuxbsd/detect.py
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: target=release_debug platform=linuxbsd pulseaudio=tru>
scons: Reading SConscript files ...
Checking for C header file mntent.h... yes
scons: done reading SConscript files.
scons: Building targets ...
^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial b>
^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial b>
^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial b>
^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial b>
^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial b>
In file included from platform/linuxbsd/display_server_x11.h:64,
                 from platform/linuxbsd/os_linuxbsd.cpp:38:
/nix/store/h9jvj8layshib3akha4i4iqhfcb8g790-libXi-1.8-dev/include/>
   33 | #include <X11/extensions/Xfixes.h> /* PointerBarrier */
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
scons: *** [platform/linuxbsd/os_linuxbsd.linuxbsd.opt.tools.64.o]>
scons: building terminated because of errors.
[Time elapsed: 00:00:04.135]
lines 1-32/32 (END)...skipping...
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/dsdcfwrj3xc4vc879yvb5yvxcy308gqg-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
applying patch /nix/store/5zm9z3bfal5asx9xbc1lw9phjrimnjya-godot-patch.patch
patching file platform/linuxbsd/detect.py
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: target=release_debug platform=linuxbsd pulseaudio=true udev=true prefix=/nix/store/l7avz2rjkhqwzxchsd7kppavm4jizmwc-godot
scons: Reading SConscript files ...
Checking for C header file mntent.h... yes
scons: done reading SConscript files.
scons: Building targets ...
^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial bui>
^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial bui>
^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] Compiling platform/linuxbsd/crash_handler_linuxbsd.cpp ...
^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial bui>
^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] ^M[Initial build] Compiling platform/linuxbsd/os_linuxbsd.cpp ...
In file included from platform/linuxbsd/display_server_x11.h:64,
                 from platform/linuxbsd/os_linuxbsd.cpp:38:
/nix/store/h9jvj8layshib3akha4i4iqhfcb8g790-libXi-1.8-dev/include/X11/extensions/XInput2.h:33:10: fatal error: X11/extensions/Xfixes.h: No such file or directory
   33 | #include <X11/extensions/Xfixes.h> /* PointerBarrier */
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
scons: *** [platform/linuxbsd/os_linuxbsd.linuxbsd.opt.tools.64.o] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:00:04.135]
```

For this one, I think it's actually a NixOS packaging bug:
```
In file included from platform/linuxbsd/display_server_x11.h:64,
                 from platform/linuxbsd/os_linuxbsd.cpp:38:
/nix/store/h9jvj8layshib3akha4i4iqhfcb8g790-libXi-1.8-dev/include/X11/extensions/XInput2.h:33:10: fatal error: X11/extensions/Xfixes.h: No such file or directory
   33 | #include <X11/extensions/Xfixes.h> /* PointerBarrier */
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
```

The error comes from XInput2.h as packaged by NixOS. It depends on Xfixes.h, so NixOS is responsible for making sure that it's package that contains XInput2.h (`xi`) also depends on the package that includes Xfixes.h.

We do parse `pkg-config xi --cflags --libs` here: https://github.com/godotengine/godot/blob/master/platform/linuxbsd/detect.py#L228
On NixOS, the CFLAGS should include the path to the Xfixes includes, as XInput2.h requires it. It's not done by upstream since it's not needed, both live in the same folder on non-NixOS distros (`/usr/include/X11/extensions/`).

--- a/platform/linuxbsd/detect.py
+++ b/platform/linuxbsd/detect.py
@@ -319,20 +319,21 @@ def configure(env):
     if os.system("pkg-config --exists alsa") == 0:  # 0 means found
         env["alsa"] = True
         env.Append(CPPDEFINES=["ALSA_ENABLED", "ALSAMIDI_ENABLED"])
+        env.ParseConfig("pkg-config alsa --cflags")  # Only cflags, we dlopen the library.
     else:
         print("Warning: ALSA libraries not found. Disabling the ALSA audio driver.")
 
     if env["pulseaudio"]:
         if os.system("pkg-config --exists libpulse") == 0:  # 0 means found
             env.Append(CPPDEFINES=["PULSEAUDIO_ENABLED"])
-            env.ParseConfig("pkg-config --cflags libpulse")
+            env.ParseConfig("pkg-config libpulse --cflags")  # Only cflags, we dlopen the library.
         else:
             print("Warning: PulseAudio development libraries not found. Disabling the PulseAudio audio driver.")
 
     if env["dbus"]:
         if os.system("pkg-config --exists dbus-1") == 0:  # 0 means found
             env.Append(CPPDEFINES=["DBUS_ENABLED"])
-            env.ParseConfig("pkg-config --cflags --libs dbus-1")
+            env.ParseConfig("pkg-config dbus-1 --cflags --libs")
         else:
             print("Warning: D-Bus development libraries not found. Disabling screensaver prevention.")
 
@@ -341,6 +342,7 @@ def configure(env):
         if env["udev"]:
             if os.system("pkg-config --exists libudev") == 0:  # 0 means found
                 env.Append(CPPDEFINES=["UDEV_ENABLED"])
+                env.ParseConfig("pkg-config libudev --cflags")  # Only cflags, we dlopen the library.
             else:
                 print("Warning: libudev development libraries not found. Disabling controller hotplugging support.")
     else:
@@ -366,11 +368,11 @@ def configure(env):
         if not env["use_volk"]:
             env.ParseConfig("pkg-config vulkan --cflags --libs")
         if not env["builtin_glslang"]:
-            # No pkgconfig file for glslang so far
+            # No pkgconfig file so far, hardcode expected lib name.
             env.Append(LIBS=["glslang", "SPIRV"])
 
     env.Append(CPPDEFINES=["GLES3_ENABLED"])
-    env.Append(LIBS=["GL"])
+    env.ParseConfig("pkg-config gl --cflags --libs")
 
     env.Append(LIBS=["pthread"])
 

